org.lflang.tests/src/org/lflang/tests/AbstractTest.java

```diff
@@ -192,4 +200,43 @@ public void runWithFourThreads() {
             true
         );
     }
+    
+    /**
+     * Swap the default runtime scheduler with other supported versions and run all the supported tests.
+     */
+    @Test
+    public void runWithNonDefaultSchedulers() {
+        Assumptions.assumeTrue(supportsSchedulerSwapping(), Message.NO_SCHED_SWAPPING_SUPPORT);
+        
+        EnumSet<TestCategory> categories = EnumSet.of(TestCategory.CONCURRENT,
+                                                      TestCategory.MULTIPORT);
+        
+        // Add federated and docker tests if supported
+        if (!isWindows() && supportsFederatedExecution()) {
+            categories.add(TestCategory.FEDERATED);
+            if (isLinux() && supportsDockerOption()) {
+                categories.add(TestCategory.DOCKER_FEDERATED);
+            }
+        }
+        
+        for (SchedulerOptions scheduler: EnumSet.allOf(SchedulerOptions.class)) {
+            if (scheduler == SchedulerOptions.getDefault()) continue;
+            
+            this.runTestsForTargets(
+                Message.DESC_SCHED_SWAPPING + scheduler.toString() +".",
+                categories::contains,
+                test -> {
+                    test.getContext()
+                        .getArgs()
+                        .setProperty(
+                                "scheduler", 
+                                scheduler.toString()
+                                ); 
+                    return true;
```

Comment:
Should the scheduler tests be run with four threads like [this](https://github.com/lf-lang/lingua-franca/blob/a1fe31566cad6f9afc834fbdd0b89abedc1f5ef0/org.lflang.tests/src/org/lflang/tests/AbstractTest.java#L235)? I only just remembered to mention it since testing seems to be so high in priority.

Reply:
I think it should be mostly okay. 

The reason that we force four threads is that multiport tests are not threaded (at least not in C, e.g., they don't specify `threads` in their target property). Also, some concurrent tests also specify one thread, which isn't particularly challenging for a scheduler. I thought setting the number of worker threads to four sounded like a good compromise.

We could also only set threads to four for the multiport test category.

Reply:
Oh, okay -- that makes sense. Thanks for explaining!